quickwit/quickwit-search/src/leaf.rs

```diff
@@ -1333,29 +1367,35 @@ pub async fn single_doc_mapping_leaf_search(
         .get_permits(permit_sizes)
         .await;
 
-    for ((split, mut request), permit_fut) in
+    let leaf_search_context = Arc::new(LeafSearchContext {
+        searcher_context: searcher_context.clone(),
+        split_outcome_counters: Arc::new(SplitSearchOutcomeCounters::new_unregistered()),
+        incremental_merge_collector: incremental_merge_collector.clone(),
+        doc_mapper: doc_mapper.clone(),
+        split_filter: split_filter.clone(),
+    });
+
+    for ((split, search_request), permit_fut) in
         split_with_req.into_iter().zip(permit_futures.into_iter())
     {
         let leaf_split_search_permit = permit_fut
             .instrument(info_span!("waiting_for_leaf_search_split_semaphore"))
             .await;
 
-        let can_be_better = check_optimize_search_request(&mut request, &split, &split_filter);
-        if !can_be_better && !run_all_splits {
+        let Some(simplified_search_request) =
+            simplify_search_request(search_request, &split, &split_filter)
+        else {
             continue;
```

Comment:
I'm not sure what split_outcome_counters does yet, but it might need to be updated here